### PR TITLE
all: update go versions to latest minor versions

### PR DESCRIPTION
**Pull request by leighmcculloch**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Update go versions to latest minor versions

### Why

Go 1.14.15 and 1.15.8 were released yesterday.

Also it looks like we are using an older version of Go in one part of the build process.

### Known limitations

N/A

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -115,8 +115,8 @@ commands:
           name: Download and install golang
           command: |
             sudo rm -rf /usr/local/go
-            wget https://dl.google.com/go/go1.14.6.linux-amd64.tar.gz
-            sudo tar -C /usr/local -xzf go1.14.6.linux-amd64.tar.gz
+            wget https://dl.google.com/go/go1.14.15.linux-amd64.tar.gz
+            sudo tar -C /usr/local -xzf go1.14.15.linux-amd64.tar.gz
```
Comment (leighmcculloch) [inline .circleci/config.yml:119]: I'm not sure why we were using such an old version of Go, looks like we haven't been keeping it updated.

File: .circleci/config.yml [hunk 2]
```diff
@@ -244,7 +244,7 @@ jobs:
   check_code_1_15:
     working_directory: /go/src/github.com/stellar/go
     docker:
-      - image: golang:1.15.7
```
Comment (bartekn) [inline .circleci/config.yml:247]: I wonder if it's possible to use [pipeline variables](https://circleci.com/docs/2.0/pipeline-variables/) here.

File: .circleci/config.yml (hunk 1 above)
Comment (bartekn) [reply .circleci/config.yml:119]: It seems this job is only used in Horizon integration tests: we start arbitrary docker containers there so we can't use golang image. As in other comment, maybe using pipeline variables will help keeping it up to date.

**Review by bartekn — APPROVED** [review]
LGTM! Since we're planning to build reproducible binaries [soon](https://github.com/stellar/go/pull/3391) can you also update the version in Jenkins?

**Comment by leighmcculloch** [comment]
>can you also update the version in Jenkins?
@stellar/horizon-committers I assume the Jenkins builds that need updating that @bartekn is referring to are for Horizon. Are you able to update them accordingly? I took a look, but your pipelines look relatively complicated and I'm not sure how to validate a change to them.

**Comment by bartekn** [comment]
> Are you able to update them accordingly?

PR opened.

**Pull request merged**